internal/infra/partition/bounds.go

```diff
@@ -42,6 +43,28 @@ func getMonthlyBounds(date time.Time) (lowerBound, upperBound time.Time) {
 	return
 }
 
+func getQuarterlyBounds(date time.Time) (lowerBound, upperBound time.Time) {
+	year, month, _ := date.Date()
+
+	var firstMonthOfTheQuarter time.Month
+
+	switch {
+	case month >= 1 && month <= 3:
+		firstMonthOfTheQuarter = time.January
+	case month >= 4 && month <= 6:
+		firstMonthOfTheQuarter = time.April
+	case month >= 7 && month <= 9:
+		firstMonthOfTheQuarter = time.July
+	case month >= 10 && month <= 12:
+		firstMonthOfTheQuarter = time.October
+	}
```

Comment:
Probably can be shorter by getting firstMonthOfTheQuarter with: 
```
quarter := (int(date.Month()) - 1) / nbMonthsInAQuarter
firstMonthOfTheQuarter := time.Month(quarter * nbMonthsInAQuarter + 1)

```

[(source)](https://blog.canopas.com/golang-date-time-utilities-part-2-b1192eb04842#1558)

Reply:
Thanks for the reference. I applied your recommendation.